### ISSUE TITLE
Incorrect Block Profit if Builder Wallet receives funds and the same block is built by builder

### ISSUE THREAD

**Issue by MPLA101** [flashbots/builder#140]
As per the documentation

* We reserve gas for the proposer payment using `proposerTxPrepare` and commit proposer payment after txs are added with
  `proposerTxCommit`. We do it in a way so all fees received by the block builder are sent to the fee recipient.

If the builder writes a block and that block has a transaction that sends funds to the Builder coin base, the whole transaction amount is considered as a block profit.

**Comment by dvush** [comment]
yes, this is one of the pitfalls to be aware of when using builder address to receive funds
